src/pybind/mgr/dashboard/frontend/src/testing/unit-test-helper.ts

```diff
@@ -40,64 +41,62 @@ export function configureTestBed(configuration, useOldMethod?) {
 }
 
 export class PermissionHelper {
-  tableActions: TableActionsComponent;
+  tac: TableActionsComponent;
   permission: Permission;
-  getTableActionComponent: () => TableActionsComponent;
 
-  constructor(permission: Permission, getTableActionComponent: () => TableActionsComponent) {
+  constructor(permission: Permission) {
     this.permission = permission;
-    this.getTableActionComponent = getTableActionComponent;
   }
 
-  setPermissionsAndGetActions(
-    createPerm: number | boolean,
-    updatePerm: number | boolean,
-    deletePerm: number | boolean
-  ): TableActionsComponent {
-    this.permission.create = Boolean(createPerm);
-    this.permission.update = Boolean(updatePerm);
-    this.permission.delete = Boolean(deletePerm);
-    this.tableActions = this.getTableActionComponent();
-    return this.tableActions;
+  setPermissionsAndGetActions(tableActions: CdTableAction[]): any {
+    const result = {};
+    [true, false].forEach((create) => {
+      [true, false].forEach((update) => {
+        [true, false].forEach((deleteP) => {
+          this.permission.create = create;
+          this.permission.update = update;
+          this.permission.delete = deleteP;
+
+          this.tac = new TableActionsComponent();
+          this.tac.selection = new CdTableSelection();
+          this.tac.tableActions = [...tableActions];
+          this.tac.permission = this.permission;
+          this.tac.ngOnInit();
+
+          result[create + '-' + update + '-' + deleteP] = {
```

Comment:
One possible solution would be to write only the name of the enabled permissions.
p.e. : 'create' and 'create-update' 
Its shorter and also easy to know what case was tested.
We could also user a ',' to separate the permissions.

Reply:
Adapted it as requested (`create,update,delete`). For the tests which don't have any permissions set, it'll create the text `no-permissions`.